mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/MimeTypeParametersDeclarationEnricher.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.internal.loader.enricher;
+
+import static org.mule.runtime.api.meta.ExpressionSupport.SUPPORTED;
+import static org.mule.runtime.api.meta.model.parameter.ParameterGroupModel.DEFAULT_GROUP_NAME;
+
+import org.mule.metadata.api.ClassTypeLoader;
+import org.mule.metadata.api.annotation.EnumAnnotation;
+import org.mule.metadata.api.model.BinaryType;
+import org.mule.metadata.api.model.StringType;
+import org.mule.metadata.api.visitor.MetadataTypeVisitor;
+import org.mule.runtime.api.meta.model.declaration.fluent.ComponentDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ExtensionDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.OperationDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterGroupDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.SourceDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.WithSourcesDeclaration;
+import org.mule.runtime.api.meta.model.display.LayoutModel;
+import org.mule.runtime.extension.api.declaration.fluent.util.IdempotentDeclarationWalker;
+import org.mule.runtime.extension.api.declaration.type.ExtensionsTypeLoaderFactory;
+import org.mule.runtime.extension.api.loader.DeclarationEnricher;
+import org.mule.runtime.extension.api.loader.ExtensionLoadingContext;
+import java.io.InputStream;
+
+/**
+ * Enriches operations which return types are {@link InputStream}, {@link String} or {@link Object} by adding two parameters:
+ * A {@link MimeTypeParametersDeclarationEnricher#MIME_TYPE_PARAMETER_NAME} that allows configuring the mimeType to
+ * the output operation payload and a {@link MimeTypeParametersDeclarationEnricher#ENCODING_PARAMETER_NAME} that allows
+ * configuring the encoding to the output operation payload.
+ * <p>
+ * Both added attributes are optional without default value and accept expressions.
+ *
+ * @since 4.0
+ */
+public final class MimeTypeParametersDeclarationEnricher implements DeclarationEnricher {
```

Comment:
it seems to me you didn't run the tests because if you don't add this into the ExtensionMOdelFactory, this is dead code